File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Hyperlinks/getshapehyperlinkdata.rb
module Asposediagramjava
  module GetShapeHyperlinkData
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Hyperlinks.vdx")

        # Get a particular shape
        shape = diagram.getPages().getPage("Flow 1").getShapes().getShape(1)

        hyperlinks = shape.getHyperlinks()

        i = 0
        while i < hyperlinks.getCount()
            hyperlink = hyperlinks.get(i)
            puts "Address: " + hyperlink.getAddress().getValue().to_s
            puts "Sub Address: " + hyperlink.getSubAddress().getValue().to_s
            puts "Description: " + hyperlink.getDescription().getValue().to_s

            i +=1
        end
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithHyperlinks/GetShapeHyperlinkData.php
<?php
namespace Aspose\Diagram\WorkingwithHyperlinks;

use com\aspose\diagram\Diagram as Diagram;
class GetShapeHyperlinkData{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram = new Diagram($dataDir . "drawing.vdx");

        # Get a particular shape
        $shape=$diagram->getPages()->getPage("Flow 1")->getShapes()->getShape(1);

        $hyperlinks=$shape->getHyperlinks();

        $i=0;
        while ($i<(int)(string)$hyperlinks->getCount()) {
            $hyperlink=$hyperlinks->get($i);
            print "Address: " . (string)$hyperlink->getAddress()->getValue();
            print "Sub Address: " . (string)$hyperlink->getSubAddress()->getValue();
            print "Description: " . (string)$hyperlink->getDescription()->getValue();
            $i+=1;
        }
    }
}